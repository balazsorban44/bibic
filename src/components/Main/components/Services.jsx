import React from 'react'
import {Link} from 'react-scroll'
import service1 from '../../../media/images/gallery/9.jpg'
import service2 from '../../../media/images/gallery/1.jpg'
import service3 from '../../../media/images/gallery/17.jpg'

const Services = () => (
  <section id="szolgaltatasok">
    <h2>Szolgáltatásaink</h2>
    <ul>
      <li>
        <img src={service1} alt="Szolgáltatás kép"/>
        <p>Klimatizált konferencia terem, bárpulttal, sörcsappal, vinotékával (50 főig).</p>
      </li>
      <li>
        <img src={service2} alt="Szolgáltatás kép"/>
        <p>Klimatizált helyiségek, fürdőszobás szobák, felszerelt konyha, tágas étkező-nappali, előadó terem.</p>
      </li>
      <li>
        <img src={service3} alt="Szolgáltatás kép"/>
        <p>Kerti konyha, kemencével, bográcshellyel, csikó tűzhellyel (rem sütő), grillezővel, füstölővel, kerti bútorral.</p>
      </li>
    </ul>
    <p>Az aktív pihenést kedvelőknek ping pong asztal és kerékpárok, a gyermekeknek játszótér áll rendelkezésére.</p>
    
    {/* <Link smooth to="galeria">Galériához</Link> */}
  </section>
)

export default Services