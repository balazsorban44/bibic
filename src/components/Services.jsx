import React from 'react'
import {Button} from './shared/Elements'
import Gallery from './shared/Gallery'
import {useTranslation} from 'react-i18next'


const Services = () => {
  const [t] = useTranslation()
  return (
    <section id="szolgaltatasok">
      <h2>{t("facilities.title")}</h2>
      <Gallery
        count={3}
        path="szolgaltatasaink"
      />
      <div className="services-footer">
        <Button
          label={t("facilities.more")}
          to="szolgaltatasok"
        />
      </div>
    </section>
  )
}

export default Services