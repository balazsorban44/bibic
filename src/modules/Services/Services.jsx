import React from "react"
import {useTranslation} from "react-i18next"

import Button from "components/Button"
import {GalleryItem} from "components/Gallery"
import Carousel from "components/Carousel"


const Services = () => {
  const [t] = useTranslation("services")
  return (
    <section id="szolgaltatasok">
      <h2>{t("title")}</h2>
      <Carousel
        className="gallery"
        count={3}
        item={GalleryItem}
        slider={false}
        type="szolgaltatasaink"
      />
      <div className="services-footer">
        <Button
          label={t("more")}
          to="szolgaltatasok"
        />
      </div>
    </section>
  )
}

export default Services