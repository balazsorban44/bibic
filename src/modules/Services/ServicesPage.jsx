import React from "react"
import {GalleryItem} from "components/Gallery"
import {useTranslation} from "react-i18next"
import useScrollTo from "hooks/useScrollTo"
import Carousel from "components/Carousel"


const ServicesPage =() => {
  const [t] = useTranslation("services")

  useScrollTo(0,0)

  return(
    <section id="szolgaltatasok">
      <h2>{t("title")}</h2>
      <Carousel
        className="gallery"
        item={GalleryItem}
        slider={false}
        type="szolgaltatasaink"
      />
    </section>
  )
}

export default ServicesPage