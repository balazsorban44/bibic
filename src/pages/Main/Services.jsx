import React from 'react'
import {Button} from '../../components/Elements'
import Gallery from '../../components/Gallery'
import {useTranslation} from 'react-i18next'


const Services = () => {
  const [t] = useTranslation()
  return (
    <section id="szolgaltatasok">
      <h2>{t("facilities.title")}</h2>
      <Gallery
        count={3}
        section="facilities"
      />
      <div className="services-footer">
        <Button
          label={t("facilities.more")}
          to="szolgaltatasok"
        />
      </div>
    </section>
  )
}

export default Services