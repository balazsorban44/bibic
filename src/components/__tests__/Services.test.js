import Services, {ServiceItem} from "../Services"

describe("Services component", () => {
  const wrapper = shallow(<Services/>)

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })

})

describe("ServiceItem component", () => {
  const props = {
    SIZE_1024: "SIZE_1024.jpg",
    SIZE_1440: "SIZE_1440.jpg",
    SIZE_640: "SIZE_640.jpg",
    desc: "desc",
    title: "title"
  }
  const wrapper = shallow(<ServiceItem {...props}/>)

  it("renders correctly", () => {
    expect(wrapper).toHaveLength(1)
  })


  it("renders description", () => {
    expect(wrapper.find("p").prop("children")).toBe(props.desc)
  })

  it("renders default img", () => {
    expect(wrapper.find("img").prop("src")).toBe(props.SIZE_640)
  })

  describe("Img", () => {
    it("alt is description", () => {
      expect(wrapper.find("img").prop("alt")).toBe(props.desc)
    })
    it("default alt is title", () => {
      wrapper.setProps({desc: ""})
      expect(wrapper.find("img").prop("alt")).toBe(props.title)
    })
  })

  describe("sources", () => {
    it("source 1", () => {
      expect(wrapper.findWhere(e => e.type() === "source" && e.prop("srcSet") === props.SIZE_1024)).toHaveLength(1)
    })

    it("source 2", () => {
      expect(wrapper.findWhere(e => e.type() === "source" && e.prop("srcSet") === props.SIZE_1440)).toHaveLength(1)
    })

  })


})