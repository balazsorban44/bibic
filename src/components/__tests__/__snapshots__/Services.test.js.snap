// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Services />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Szolgáltatásaink
        </h2>,
        <withRouter(Component)
          count={3}
          item={[Function]}
          path="szolgaltatasaink"
        />,
        <div
          className="services-footer"
        >
          <p>
            Az aktív pihenést kedvelőknek ping pong asztal és kerékpárok, a gyermekeknek játszótér áll rendelkezésére.
          </p>
          <Button
            label="További szolgáltatásaink →"
            to="szolgaltatasok"
          />
        </div>,
      ],
      "id": "szolgaltatasok",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Szolgáltatásaink",
        },
        "ref": null,
        "rendered": "Szolgáltatásaink",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "count": 3,
          "item": [Function],
          "path": "szolgaltatasaink",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Az aktív pihenést kedvelőknek ping pong asztal és kerékpárok, a gyermekeknek játszótér áll rendelkezésére.
            </p>,
            <Button
              label="További szolgáltatásaink →"
              to="szolgaltatasok"
            />,
          ],
          "className": "services-footer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Az aktív pihenést kedvelőknek ping pong asztal és kerékpárok, a gyermekeknek játszótér áll rendelkezésére.",
            },
            "ref": null,
            "rendered": "Az aktív pihenést kedvelőknek ping pong asztal és kerékpárok, a gyermekeknek játszótér áll rendelkezésére.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "label": "További szolgáltatásaink →",
              "to": "szolgaltatasok",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Szolgáltatásaink
          </h2>,
          <withRouter(Component)
            count={3}
            item={[Function]}
            path="szolgaltatasaink"
          />,
          <div
            className="services-footer"
          >
            <p>
              Az aktív pihenést kedvelőknek ping pong asztal és kerékpárok, a gyermekeknek játszótér áll rendelkezésére.
            </p>
            <Button
              label="További szolgáltatásaink →"
              to="szolgaltatasok"
            />
          </div>,
        ],
        "id": "szolgaltatasok",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Szolgáltatásaink",
          },
          "ref": null,
          "rendered": "Szolgáltatásaink",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "count": 3,
            "item": [Function],
            "path": "szolgaltatasaink",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Az aktív pihenést kedvelőknek ping pong asztal és kerékpárok, a gyermekeknek játszótér áll rendelkezésére.
              </p>,
              <Button
                label="További szolgáltatásaink →"
                to="szolgaltatasok"
              />,
            ],
            "className": "services-footer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Az aktív pihenést kedvelőknek ping pong asztal és kerékpárok, a gyermekeknek játszótér áll rendelkezésére.",
              },
              "ref": null,
              "rendered": "Az aktív pihenést kedvelőknek ping pong asztal és kerékpárok, a gyermekeknek játszótér áll rendelkezésére.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "label": "További szolgáltatásaink →",
                "to": "szolgaltatasok",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
